studio/localStores/storageExplorer/StorageExplorerStore.js

```diff
@@ -993,11 +1007,32 @@ class StorageExplorerStore {
     this.openedFolders = updatedOpenedFolders
   }
 
+  // Check parent folder if its empty, if yes, reinstate .emptyFolderPlaceholder
+  // Used when deleting folder or deleting files
+  validateParentFolderEmpty = async (parentFolderPrefix) => {
+    const { data: items, error } = await this.supabaseClient.storage
+      .from(this.selectedBucket.name)
+      .list(parentFolderPrefix, this.DEFAULT_OPTIONS)
+    if (!error && items.length === 0) {
+      const prefixToPlaceholder = `${parentFolderPrefix}/${EMPTY_FOLDER_PLACEHOLDER_FILE_NAME}`
+      await this.supabaseClient.storage
+        .from(this.selectedBucket.name)
+        .upload(prefixToPlaceholder, new File([], EMPTY_FOLDER_PLACEHOLDER_FILE_NAME))
+    }
+  }
+
   deleteFolder = async (folder) => {
+    const isDeleteFolder = true
     const files = await this.getAllItemsAlongFolder(folder)
-    const showFilesDeleteToast = false
+    await this.deleteFiles(files, isDeleteFolder)
+
+    const parentFolderPrefix = this.openedFolders.map((folder) => folder.name).join('/')
```

Comment:
the empty folder validation has to run separately from that within deleteFiles, and deleteFolder only need to check it's parent folder, and not the parent folder of each individual file within the folder